util/reggen/validate.py

```diff
@@ -522,24 +557,26 @@ def bitmask(bfield):
 
 # swaccess permitted values
 # text description, access enum, wr access enum, rd access enum, ok in window
+# flake8 doens't support noqa for a block.
+# So should choose between visually hard-to-redable and noqa comment every lines.
 swaccess_permitted = {
-    'ro':    ("Read Only",
-                        SwAccess.RO,  SwWrAccess.NONE, SwRdAccess.RD,   True),
-    'rc':    ("Read Only, reading clears",
-                        SwAccess.RC,  SwWrAccess.WR,   SwRdAccess.RC,   False),
-    'rw':    ("Read/Write",
-                        SwAccess.RW,  SwWrAccess.WR,   SwRdAccess.RD,   True),
-    'r0w1c': ("Read zero, Write with 1 clears",
-                        SwAccess.W1C, SwWrAccess.WR,   SwRdAccess.NONE, False),
-    'rw1s':  ("Read, Write with 1 sets",
-                        SwAccess.W1S, SwWrAccess.WR,   SwRdAccess.RD,   False),
-    'rw1c':  ("Read, Write with 1 clears",
-                        SwAccess.W1C, SwWrAccess.WR,   SwRdAccess.RD,   False),
-    'rw0c':  ("Read, Write with 0 clears",
-                        SwAccess.W0C, SwWrAccess.WR,   SwRdAccess.RD,   False),
-    'wo':    ("Write Only",
-                        SwAccess.WO,  SwWrAccess.WR,   SwRdAccess.NONE, True)
-} # yapf: disable
+    'ro':    ("Read Only",                                             # noqa: E241
+              SwAccess.RO,  SwWrAccess.NONE, SwRdAccess.RD,   True),   # noqa: E241
+    'rc':    ("Read Only, reading clears",                             # noqa: E241
+              SwAccess.RC,  SwWrAccess.WR,   SwRdAccess.RC,   False),  # noqa: E241
+    'rw':    ("Read/Write",                                            # noqa: E241
+              SwAccess.RW,  SwWrAccess.WR,   SwRdAccess.RD,   True),   # noqa: E241
+    'r0w1c': ("Read zero, Write with 1 clears",                        # noqa: E241
+              SwAccess.W1C, SwWrAccess.WR,   SwRdAccess.NONE, False),  # noqa: E241
+    'rw1s':  ("Read, Write with 1 sets",                               # noqa: E241
+              SwAccess.W1S, SwWrAccess.WR,   SwRdAccess.RD,   False),  # noqa: E241
+    'rw1c':  ("Read, Write with 1 clears",                             # noqa: E241
+              SwAccess.W1C, SwWrAccess.WR,   SwRdAccess.RD,   False),  # noqa: E241
+    'rw0c':  ("Read, Write with 0 clears",                             # noqa: E241
+              SwAccess.W0C, SwWrAccess.WR,   SwRdAccess.RD,   False),  # noqa: E241
+    'wo':    ("Write Only",                                            # noqa: E241
+              SwAccess.WO,  SwWrAccess.WR,   SwRdAccess.NONE, True)    # noqa: E241
```

Comment:
This is quite annoying part of using flake8, I would like to turn off E241 error for this entire block but not able to do that. Either seeing ugly aligned code or ugly noqa comment in every lines.

Reply:
Yeah, I agree. It's a pity that flake8 doesn't have a block version of noqa. That said, I think this code probably looks ok (since the ugly noqa comments are at least "tabulated").